src/api/posts.js

```diff
@@ -0,0 +1,17 @@
+import { BASE_URL } from './api';
+
+export function getUserPosts(userId) {
+  const queryString = +userId === 0
+    ? `${BASE_URL}/posts/`
+    : `${BASE_URL}/posts?userId=${userId}`;
+
+  return fetch(queryString)
+    .then(response => response.json())
+    .then(response => response.data);
+}
+
+export function getPostDetails(postId) {
+  return fetch(`${BASE_URL}/posts/${postId}`)
+    .then(response => response.json())
+    .then(response => response.data);
```

Comment:
you can create a common function getData and move fetch with 2 then there
https://mate-academy.github.io/fe-program/js/extra/fetch